05_stringer/undrewb/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	want := strconv.Quote("127.0.0.1\n")
+	got := strconv.Quote(buf.String())
+
+	assert.Equal(t, got, want)
+}
+
+var stringerData = []struct {
+	name  string
+	input IPAddr
+	want  string
+}{
+	{name: "localhost",
+		input: IPAddr{127, 0, 0, 1},
+		want:  "127.0.0.1"},
+	{name: "four octet addr",
+		input: IPAddr{68, 2, 44, 125},
+		want:  "68.2.44.125"},
+	{name: "three octet addr",
+		input: IPAddr{68, 2, 125},
+		want:  "68.2.125.0"},
+	{name: "two octet addr",
+		input: IPAddr{9, 5},
+		want:  "9.5.0.0"},
+	{name: "one octet addr",
+		input: IPAddr{144},
+		want:  "144.0.0.0"},
+	{name: "empty",
+		input: IPAddr{},
+		want:  "0.0.0.0"},
+}
+
+func TestStringer(t *testing.T) {
+	for _, tt := range stringerData {
```

Comment:
Much cleaner:
```go
func TestIPAddrStringer(t *testing.T) {
	assertString := func(name string, expected string, input IPAddr) {
		t.Run(name, func(t *testing.T) {
			assert.Equal(t, expected, fmt.Sprint(input))
		})
	}

	assertString("localhost", "127.0.0.1", IPAddr{127, 0, 0, 1})
	assertString("four octet addr", "68.2.44.125", IPAddr{68, 2, 44, 125})
	⋮
}
```

Reply:
Disagree, @anzdaddy - @undrewb 's code is the typical table driven test look and feel.

Reply:
leaving table driven test format